From 90a8dc30c23cdfeec4780716c486c09203ec5e65 Mon Sep 17 00:00:00 2001
From: James <jbean46@gmail.com>
Date: Tue, 18 Sep 2018 12:02:18 -0400
Subject: [PATCH 01/66] Remove redundant import

---
 .../PitchSpeller/Wetherfield/TendencyConverting.swift            | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Sources/SpelledPitch/PitchSpeller/Wetherfield/TendencyConverting.swift b/Sources/SpelledPitch/PitchSpeller/Wetherfield/TendencyConverting.swift
index 198be7e..40d8370 100644
--- a/Sources/SpelledPitch/PitchSpeller/Wetherfield/TendencyConverting.swift
+++ b/Sources/SpelledPitch/PitchSpeller/Wetherfield/TendencyConverting.swift
@@ -6,7 +6,6 @@
 //
 
 import Pitch
-import SpelledPitch
 
 /// Interface for `PitchSpellingCategoryProtocol` types which can convert a `TendencyPair` into a
 /// `ModifierDirection`.
-- 
2.17.2 (Apple Git-113)

