From a4bf47b93b40e73800bc22f070309d5c1a5fab9a Mon Sep 17 00:00:00 2001
From: James Bean <jbean46@gmail.com>
Date: Mon, 1 Oct 2018 15:50:03 -0400
Subject: [PATCH 16/66]  Flow tests (#116)

---
 Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift b/Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift
index 9d9efca..9cb5a46 100644
--- a/Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift
+++ b/Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift
@@ -27,6 +27,8 @@ extension FlowNetwork {
     /// Create a `FlowNetwork` with the given `directedGraph` and the given `source` and `sink` nodes.
     init(_ directedGraph: WeightedDirectedGraph<Node,Weight>, source: Node, sink: Node) {
         self.directedGraph = directedGraph
+        self.directedGraph.insert(source)
+        self.directedGraph.insert(sink)
         self.source = source
         self.sink = sink
     }
-- 
2.17.2 (Apple Git-113)

