From 9ed2fa3bf3b4e79f23e4abcee41fb8eba7a99dba Mon Sep 17 00:00:00 2001
From: James <jbean46@gmail.com>
Date: Thu, 18 Oct 2018 10:49:29 -0400
Subject: [PATCH 18/66] Rename .flat(Int) -> .flat(count: Int)

---
 .../Wetherfield/Pitch.Spelling.Category.swift        | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Sources/SpelledPitch/PitchSpeller/Wetherfield/Pitch.Spelling.Category.swift b/Sources/SpelledPitch/PitchSpeller/Wetherfield/Pitch.Spelling.Category.swift
index 1513197..29ccf06 100644
--- a/Sources/SpelledPitch/PitchSpeller/Wetherfield/Pitch.Spelling.Category.swift
+++ b/Sources/SpelledPitch/PitchSpeller/Wetherfield/Pitch.Spelling.Category.swift
@@ -23,42 +23,42 @@ extension Pitch.Spelling {
         /// Category for pitch classes `0` and `5`.
         struct Zero: PitchSpellingCategoryProtocol {
             static var modifiers: ModifierMap {
-                return [.down: .flat(2), .neutral: .natural, .up: .sharp(count: 1)]
+                return [.down: .flat(count: 1), .neutral: .natural, .up: .sharp(count: 1)]
             }
         }
 
         /// Category for pitch classes `1` and `6`.
         struct One: PitchSpellingCategoryProtocol {
             static var modifiers: ModifierMap {
-                return [.down: .flat(1), .neutral: .sharp(count: 1), .up: .sharp(count: 2)]
+                return [.down: .flat(count: 1), .neutral: .sharp(count: 1), .up: .sharp(count: 2)]
             }
         }
 
         /// Category for pitch classes `2`, `7`, and `9`.
         struct Two: PitchSpellingCategoryProtocol {
             static var modifiers: ModifierMap {
-                return [.down: .flat(2), .neutral: .natural, .up: .sharp(count: 2)]
+                return [.down: .flat(count: 1), .neutral: .natural, .up: .sharp(count: 2)]
             }
         }
 
         /// Category for pitch classes `3`, and `10`.
         struct Three: PitchSpellingCategoryProtocol {
             static var modifiers: ModifierMap {
-                return [.down: .flat(2), .neutral: .flat(1), .up: .sharp(count: 1)]
+                return [.down: .flat(count: 1), .neutral: .flat(count: 1), .up: .sharp(count: 1)]
             }
         }
 
         /// Category for pitch classes `4`, and `11`.
         struct Four: PitchSpellingCategoryProtocol {
             static var modifiers: ModifierMap {
-                return [.down: .flat(1), .neutral: .natural, .up: .sharp(count: 2)]
+                return [.down: .flat(count: 1), .neutral: .natural, .up: .sharp(count: 2)]
             }
         }
 
         /// Category for pitch class `8`.
         struct Five: PitchSpellingCategoryProtocol {
             static var modifiers: ModifierMap {
-                return [.down: .flat(1), .up: .sharp(count: 1)]
+                return [.down: .flat(count: 1), .up: .sharp(count: 1)]
             }
         }
 
-- 
2.17.2 (Apple Git-113)

