From 69b87875d01671e4e466afd886dec564144bae0d Mon Sep 17 00:00:00 2001
From: bwetherfield <bwetherfield@gmail.com>
Date: Sun, 4 Nov 2018 01:41:52 -0400
Subject: [PATCH 36/66] Make switch style clearer (#157)

---
 .../SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift   | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift b/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift
index 63aefb6..e1820d5 100644
--- a/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift
+++ b/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift
@@ -34,8 +34,8 @@ struct PitchSpeller {
     )
 
     static let adjacencyCarrying = AdjacencyCarrying.build(from: PitchSpeller.tendencyGraph)
-    static let tendencyMask: AdjacencyCarrying<Graph<FlowNode<Cross<Int,Tendency>>>> = adjacencyCarrying.pullback {
-        switch $0 {
+    static let tendencyMask: AdjacencyCarrying<Graph<FlowNode<Cross<Int,Tendency>>>> = adjacencyCarrying.pullback { node in
+        switch node {
         case .source: return .down
         case .sink: return .up
         case .`internal`(let index): return index.b
-- 
2.17.2 (Apple Git-113)

