From 1f2a098a05be3a53bab687a84a3c012f18bbf05e Mon Sep 17 00:00:00 2001
From: James Bean <jbean46@gmail.com>
Date: Fri, 16 Nov 2018 21:46:22 -0500
Subject: [PATCH 62/66] Conform FlowNode to CustomStringConvertible (#204)

---
 .../PitchSpeller/Wetherfield/Wetherfield.swift  | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift b/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift
index c7af8ce..fc63f87 100644
--- a/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift
+++ b/Sources/SpelledPitch/PitchSpeller/Wetherfield/Wetherfield.swift
@@ -334,3 +334,20 @@ extension WeightedDirectedGraph where Weight: ExpressibleByIntegerLiteral {
         }
     }
 }
+
+extension FlowNode: CustomStringConvertible {
+
+    // MARK: - CustomStringConvertible
+
+    /// Printable description of `FlowNode`.
+    public var description: String {
+        switch self {
+        case .source:
+            return "source"
+        case .sink:
+            return "sink"
+        case .internal(let index):
+            return "\(index)"
+        }
+    }
+}
-- 
2.17.2 (Apple Git-113)

